src/main/java/com/zendesk/maxwell/RunLoopProcess.java

```diff
@@ -31,8 +34,8 @@ public boolean runLoop() throws Exception {
 			while (this.runState == RunState.RUNNING)
 				work();
 
-			this.beforeStop();
 		} finally {
+			this.beforeStop();
```

Comment:
just curiosity, not too sure on finally behavior =), if `beforeStop` throws `runState` won't be changed? (was going to suggest inverting the statements order, but afaik won't matter much, and the program will just finish main execution)